<launch>
  <!-- Unreal/Routine arguments -->
  <arg name="sensor_model_config" default="sensor_model_config.yaml"/>



  <!-- Unreal vision client -->
  <node name="unreal_ros_client" pkg="unreal_cv_ros" type="unreal_ros_client.py" output="screen">
    <rosparam param="mode">test</rosparam>
  </node>
  
  <!-- Unreal sensor model -->
  <node name="unreal_sensor_model" pkg="unreal_cv_ros" type="sensor_model.py" output="screen">
    <rosparam command="load" file="$(find unreal_cv_ros)/cfg/$(arg sensor_model_config)"/>
    <param name="camera_params_ns" value="unreal_ros_client/camera_params"/>
    <remap from="ue_sensor_raw" to="unreal_ros_client/ue_sensor_raw" />
  </node>

  <!-- RVIZ -->
  <node type="rviz" name="rviz_pcl" pkg="rviz" args="-d $(find unreal_cv_ros)/cfg/example_test.rviz"/>
  
  <!-- static camera transform (odom to camera frame)-->
   <node pkg="tf" type="static_transform_publisher" name="tf_camera_to_link" args="0 0 0 -0.5 0.5 -0.5 0.5 /camera_link /camera 100"/>
   
</launch>
